ruby/ql/test/library-tests/dataflow/params/params_flow.rb

```diff
@@ -57,4 +57,10 @@ def posargs(p1, *posargs)
 posargs(taint(23), *args)
 
 args = [taint(24), taint(25)]
-posargs(*args)
\ No newline at end of file
+posargs(*args)
+
+args = taint(26)
+def splatstuff(*x)
+    sink x # $ hasValueFlow=26
```

Comment:
I don't think we want flow in this case; we only want flow to `x[n]`.

Reply:
I've made it similar to how we handle hash-splat, except I still restrict it to the first argument/parameter.  

It still gives me the flow I want in [the second-order-command-injection PR](https://github.com/github/codeql/pull/11236).